### ISSUE TITLE
Text Notes on Brew Days gone and do not work anymore

### ISSUE THREAD

**Issue by cgspeck** [Brewtarget/brewtarget#457]
Before #453 I was able to store notes against each brew day.

After upgrading all the notes have gone blank. If I attempt to save a note, then an error is thrown and the application crashes:
![image](https://user-images.githubusercontent.com/1311012/78756163-b2935000-79bd-11ea-992d-30f125f1d94c.png)

```
[17:22:55.327] ERROR : Could not translate notes to a column name
Qt has caught an exception thrown from an event handler. Throwing
exceptions from an event handler is not supported in Qt.
You must not let any exception whatsoever propagate through Qt code.
If that is not possible, in Qt 5 you must at least reimplement
QCoreApplication::notify() and catch all exceptions there.
```

Steps to reproduce:

1. click on a recipe
2. click "Brew It" or select an existing brew day
3. click on the relevant date tab, then enter any text in `Notes` box
4. click away from the notes box

The schema of my brewnotes table is this:

```
CREATE TABLE "brewnote" 
(
    id INTEGER PRIMARY KEY autoincrement ,
    abv real DEFAULT 0,
    attenuation real DEFAULT 1,
    boil_off real DEFAULT 1,
    brewdate timestamp DEFAULT CURRENT_TIMESTAMP,
    brewhouse_eff real DEFAULT 70,
    deleted boolean DEFAULT 0,
    display boolean DEFAULT 1,
    eff_into_bk real DEFAULT 70,
    fermentdate timestamp DEFAULT CURRENT_TIMESTAMP,
    fg real DEFAULT 1,
    final_volume real DEFAULT 1,
    folder text DEFAULT '',
    mash_final_temp real DEFAULT 67,
    og real DEFAULT 1,
    pitch_temp real DEFAULT 20,
    post_boil_volume real DEFAULT 0,
    projected_abv real DEFAULT 1,
    projected_atten real DEFAULT 75,
    projected_boil_grav real DEFAULT 1,
    projected_eff real DEFAULT 1,
    projected_ferm_points real DEFAULT 1,
    projected_fg real DEFAULT 1,
    projected_mash_fin_temp real DEFAULT 67,
    projected_og real DEFAULT 1,
    projected_points real DEFAULT 1,
    projected_strike_temp real DEFAULT 70,
    projected_vol_into_bk real DEFAULT 1,
    projected_vol_into_ferm real DEFAULT 0,
    sg real DEFAULT 1,
    strike_temp real DEFAULT 70,
    volume_into_bk real DEFAULT 0,
    volume_into_fermenter real DEFAULT 0,
    recipe_id integer,
    FOREIGN KEY(recipe_id) REFERENCES recipe(id)
)
```

If I add a `TEXT` field called `notes` to the `brewnote` table, then the contents of this field will show up in the Notes box, but if I attempt to add any of it or edit it, then the error above happens and Brewtarget crashes.

Notes on recipes still appear to work.

**Comment by mikfire** [comment]
Ugh. This is a bad combo.

The root cause here is that I did not include the notes column in the TableProperties for brewnotes. As part of the upgrade, I dropped two columns from brewnotes that were not used. 

On SQLite, dropping a column requires making a temporary table, copying the data from the old to the temp, dropping the old and renaming the temp. Since I didn't define the notes column, the notes weren't copied.

I am fixing the code and will have a pull request later today. 

I am not sure if you lost data or not. I would think you have, but you seem to indicate the data is still there, just not available until you modify the table.

If you didn't lose data, then my later PR will fix the remaining issues.

If you did lose data, you will need to:
 1.  stop brewtarget 
 2. Find your DATA_DIR (`~/.config/brewtarget` for me, but yours may be somewhere else). 
 3. Copy database.sqlite to somewhere safe, eg, `cp database.sqlite /var/tmp` 
 4. There should be a mess of backup copies named `bt_database.[datestamp]`. Find the one timestamped to closest before you ran the new code. Alternately, find the most recent one that shows settings::version to be 7.
  5. Verify you followed step three
  6. Copy the `bt_database.[datastamp]` you found in step 4 to database.sqlite, eg `cp bt_database.20200303 database.sqlite`
  7. Start brewtarget
That will fire the database upgrade again, but this time without forgetting about brewnotes notes.
